dapps/marketplace/src/hoc/withTokenBalance.js

```diff
@@ -11,7 +11,9 @@ function withTokenBalance(WrappedComponent) {
       variables={{ account: wallet, token }}
       skip={!wallet}
     >
-      {({ data }) => <WrappedComponent {...props} {...tokenInfo(data)} />}
+      {({ data }) => (
+        <WrappedComponent wallet={wallet} {...props} {...tokenInfo(data)} />
```

Comment:
Why do we need this extra prop? Seems unrelated to what this hoc is doing?

Reply:
`ListingDetail` component uses `withWallet` HOC but the `wallet` prop is not passed down when this HOC is used since ES6 destructing is used to extract the `wallet` and `token` from other props.

I needed both `wallet` and `walletProxy` props inside `ListingDetail` component. In old listings created without a proxy, we also have to use `wallet` prop to check if the current user created the listing in addition to `walletProxy`.

Reply:
Ah makes sense